docs/preview/yamlgettingstarted-triggers.md

```diff
@@ -0,0 +1,96 @@
+# YAML getting started - YAML triggers (not yet available, for discussion only)
+
+## Simple CI trigger syntax
+
+On push, the YAML file from the branch that was updated, is used to determine whether
+to queue a CI build.
+
+For example, if master is updated, the trigger defined in master will be used. If a different
+branch is updated, the YAML file from that branch will be used to determine whether to queue a
+build.
+
+```yaml
+trigger:
+- master
+- releases/*
+```
+
+## CI is opt-out
+
+When not specified, the default trigger for build would be all branches.
+
+For build this behavior makes sense. Especially when we start supporting on-push defintion creation from any branch. And on-push resource authorization from any branch.
+
+For release, CI trigger would likely be opt-in? However with YAML, release will be tightly coupled with a repository. So opt-out CI behavior might also make sense for release? Or opt-out, but the default for RM only includes master?
+
+```yaml
+trigger: false | none # Turn off CI
+```
+
+## Full CI trigger syntax
+
+```yaml
+trigger:
+  batch: bool | number
+  branches:
+    include: []
+    exclude: []
+  paths:
+    include: []
+    exclude: []
+```
+
+## Schedules (future; needs cleanup)
+
+```yaml
+schedules:
+- schedule: name
+  whenUnchanged: bool
+  days: # Also supports a string instead of an array: "<DAY>" or "All" or "Weekdays"
+  - sun
+  - mon
+  - tues
+  - wed
+  - thurs
+  - fri
+  - sat
+  timeZone: EST # Defaults to account time zone? Do we know that info? We might need to maintain a mapping for this abbrev, it is not in system.timeZoneInfo.
```

Comment:
Is there any way you could reference a default value for an entire collection so that you could easily ensure everyone is using the same time zone?  ie, I don't care what the timezone is, I just want it to be consistent with the rest of devdiv.visualstudio.com

Reply:
This sort of goes into a general templating thing...I could define a standard dotnet core template for periodic builds, supply a parameter or two for specific info, and have the template take care of time/zone.  e.g.

`- standardNetCorePeriodicBuild([monday, friday])`

Which goes to:

```
- schedules:
  - 1:00am
- whenUnchanged: true
- timeZone: PST
- days:
  - <parameter from input on standardNetCorePeriodicBuild>
```

Reply:
Yeah, templating would be one way.  something like `-timeZone: $(ProjectCollection.TimeZone)` is another, though that could be in a template as well.

Reply:
by default not specifying a timeZone is going to use the timeZone configured for your account.

Yes templating is also something we need to look at here.

Reply:
@chrisrpatterson We were talking about overall build tooling (e.g. msbuild tasks) today, and versioning these build tools comes up a lot.  We version them today through the use of nuget packages.  I would be interested to know what kind of thoughts you have on versioning templates. It would be great if templates and other product tooling could be treated similarly (e.g. nuget packages that projects pull updates to as needed)

Reply:
@mmitche lets take this topic discussion to a separate thread.